app/models/manageiq/providers/vmware/discovery.rb

```diff
@@ -4,70 +4,61 @@ module ManageIQ
   module Providers
     module Vmware
       class Discovery
-        SERVER_CONSOLE_PORTS = [902, 912].freeze
-
-        ESX_PORTS = [902, 903].freeze
-
-        VC_PORTS = [
-          [
-            135, # VC < 5.1 or
-            7444 # VC >= 5.1
-          ],
-          [
-            139,  # VC < 5.1 or
-            2012, # VC >= 5.1
-            2013,
-            2014
-          ]
-        ].freeze
-
         def self.probe(ost)
-          # Check if VMware Server
-          ost.hypervisor << :vmwareserver if ManageIQ::NetworkDiscovery::Port.scan_open(ost, SERVER_CONSOLE_PORTS).length == 2
-
-          # First check if we can access the VMware webservice before even trying the port scans.
-          begin
-            require 'VMwareWebService/MiqVimClientBase'
-            MiqVimClientBase.new(ost.ipaddr, "test", "test")
-          rescue => err
-            $log&.debug("Vmware::Discovery: Failed to connect to VMware webservice: #{err}. ip = #{ost.ipaddr}")
-            return
+          # Get Info from VMware (vSphere) Web Service API
+          access_webservice(ost) do |info|
+            if add_hypervisor(ost, info)
```

Comment:
I feel like add_hypervisor and add_os are doing too much.
I would rather see something like:
```
hvisor_type = hypervisor_type(info)
if ost.discover_types.include?(hvisor_type)
  ost.hypervisor << hvisor_type
  ost.os << hvisor_os_type(info)
end
```